### PR TITLE
enable proposer preferences ref tests

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Enable proposer preferences gossip ref tests

This PR requires SSZ ref tests to be fixed first

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Medium Risk**
> Changes proposer-preferences gossip acceptance rules and dependent-root fork-choice logic on the Gloas path; behavior is spec-driven but affects what messages nodes accept on the network.
> 
> **Overview**
> Turns on **`networking/gossip_proposer_preferences`** consensus ref tests by bumping the stable vectors to **`v1.7.0-beta.1`** and wiring a **`GossipProposerPreferencesTestExecutor`** that builds the fixture chain with Gloas schemas and a Gloas **`setupSpec`**, while still validating gossip against the test’s configured spec.
> 
> **Proposer preferences gossip** now rejects dependent blocks whose slot is **after the shuffling-dependent slot** (via **`ShufflingDependentRootUtil`**) instead of comparing against the proposer lookahead epoch start; blocks **at** that slot are allowed.
> 
> **`GossipValidationHelper.isPossibleDependentRoot`** drops the check that immediately ruled out roots at or after the epoch start slot, so a **genesis head** can be a valid dependent root for the genesis epoch—needed for ref tests and edge cases.
> 
> Unit tests were updated to match the new rejection/accept rules and the genesis dependent-root behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 368b02d1db7a1a1a30e1ac1257b9544466eccb8f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipProposerPreferencesTestExecutor.java [hunk 1]
```diff
@@ -47,7 +49,14 @@ public void runTest(final TestDefinition testDefinition) throws Throwable {
         loadYaml(testDefinition, "meta.yaml", GossipProposerPreferencesMetaData.class);
     final boolean signatureVerificationDisabled = metaData.getBlsSetting() == BlsSetting.IGNORED;
     final Spec spec = testDefinition.getSpec(!signatureVerificationDisabled);
-    final BeaconState state = loadStateFromSsz(testDefinition, "state.ssz_snappy");
+    final SchemaDefinitionsGloas schemaDefinitions =
+        SchemaDefinitionsGloas.required(
+            spec.forMilestone(SpecMilestone.GLOAS).getSchemaDefinitions());
```
Comment (tbenr) [inline eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipProposerPreferencesTestExecutor.java]: i'd use testDefinition's milestone instead of hardcoding GLOAS here (and in other places) is more future proof

Comment (mehdi-aouadi) [reply eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/gossip/GossipProposerPreferencesTestExecutor.java]: good catch, fixed

**Review by tbenr — APPROVED** [review]
